src/pages/signIn/SignIn.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { Formik } from 'formik';
+
+import { RegularText, SignForm, SignInButton, StyledLink, StyledTitle, SignInForm } from './styled';
+
+import { getUserInfo } from '../../redux/actions/actionCreators';
+import { initValuesForLogIn, ValidSchemeForLogIn } from './validation';
+import { useQueryToLogin } from '../../hooks/query/loginQuery';
+import { fetchUsers } from '../../services/userService';
+import ValidInput from './validInput';
+
+const SignIn = () => {
+    const [userCredentials, setUserCredentials] = useState({});
+    const dispatch = useDispatch();
+
+    const { data } = useQueryToLogin('currentUser', () => fetchUsers(), {
+        enabled: !!userCredentials,
+    });
+
+    const singInHandler = ({ email, password }) => {
+        const currentUser = data.data.find(
+            (dataElement) => dataElement.email == email && dataElement.password == password
+        );
+        if (currentUser) {
+            dispatch(getUserInfo(currentUser));
+            return true;
+        }
+        setUserCredentials({});
+        return false;
```

Comment:
you don't need to pass false when you need to return false. you can just stay with "return"

Reply:
and actually you can just make 1 return like "return Boolean(currentUser)"